telethon_generator/tl_generator.py

```diff
@@ -687,21 +687,17 @@ def write_request_result_code(builder, tlobject):
                          will be written
         """
         if tlobject.result.startswith('Vector<'):
-            # Vector results are a bit special since they can also be composed
-            # of integer values and such; however, the result of requests is
-            # not parsed as arguments are and it's a bit harder to tell which
-            # is which.
-            if tlobject.result == 'Vector<int>':
+            # Check if Vector contents are boxed or not
+            # We're implying that constructor names always
+            # start with lowecase and types start with uppercase
+            match = re.match(r'Vector<(.*)>', tlobject.result)
+            if match and match.group(1).lower:
```

Comment:
You're not calling `.lower()`. This will always be `True`, as `bool(func) == True`.

Reply:
Also you probably want to check for `match.group(1)[0]`, as `ipPort` is valid and has upper case letters.

Reply:
Oh lol

Reply:
Ruby :(